tests/test_buttons.py

```diff
@@ -9,52 +53,93 @@ def test_tactile_buttons_initialization(mocker, m5stickv):
+    # Assert button methods return default values
+    assert button.value() == RELEASED
+    assert not button.event()
+    mock_tactile_buttons.event_handler.assert_not_called()
 
 
-def test_button_page_event(mocker, m5stickv):
-    from krux.buttons import ButtonPage, buttons_control
+def test_no_button_enter(mocker, m5stickv, mock_tactile_buttons):
+    from krux.buttons import ButtonEnter, RELEASED
 
-    # Simulate button page pin
-    mocked_pin = 11
-    pin = mocked_pin
-    button_page = ButtonPage(pin)
-    # Overwrite the pin's MagicMock with the mocked pin number
-    buttons_control.page = mocked_pin
+    # Simulate device has no button enter
+    mocker.patch.object(mock_tactile_buttons, "init_enter", lambda pin: None)
+    mock_tactile_buttons.enter = None
+    button_enter = ButtonEnter(None)
+    assert button_enter.value() == RELEASED
+    assert not button_enter.event()
+    mock_tactile_buttons.event_handler.assert_not_called()
 
-    # Simulate button press
-    buttons_control.event_handler(pin)
-    assert button_page.event()
-    # Assert that the event flag is reset
-    assert not button_page.event()
 
+def test_no_button_page(mocker, m5stickv, mock_tactile_buttons):
+    from krux.buttons import ButtonPage, RELEASED
 
-def test_button_page_prev_event(mocker, m5stickv):
-    from krux.buttons import ButtonPagePrev, buttons_control
+    # Simulate device has no button page
+    mocker.patch.object(mock_tactile_buttons, "init_page", lambda pin: None)
+    mock_tactile_buttons.page = None
+    button_page = ButtonPage(None)
+    assert button_page.value() == RELEASED
+    assert not button_page.event()
+    mock_tactile_buttons.event_handler.assert_not_called()
 
-    # Simulate button page pin
-    mocked_pin = 12
-    pin = mocked_pin
-    button_page_prev = ButtonPagePrev(pin)
-    # Overwrite the pin's MagicMock with the mocked pin number
-    buttons_control.page_prev = mocked_pin
 
-    # Simulate button press
-    buttons_control.event_handler(pin)
-    assert button_page_prev.event()
-    # Assert that the event flag is reset
+def test_no_button_no_page_prev(mocker, m5stickv, mock_tactile_buttons):
+    from krux.buttons import ButtonPagePrev, RELEASED
+
+    # Simulate device has no button page
+    mocker.patch.object(mock_tactile_buttons, "init_page_prev", lambda pin: None)
+    mock_tactile_buttons.page_prev = None
+    button_page_prev = ButtonPagePrev(None)
+    assert button_page_prev.value() == RELEASED
     assert not button_page_prev.event()
+    mock_tactile_buttons.event_handler.assert_not_called()
+
+
+def test_button_enter(mocker, m5stickv, mock_tactile_buttons):
+    from krux.buttons import ButtonEnter
+
+    button = ButtonEnter(mock_tactile_buttons.enter)
+
+    # Simulate button pressed
+    mock_tactile_buttons.event_handler(mock_tactile_buttons.enter)
+
+    # if really pressed, should be True
+    assert button.event()
+
+    # if released, should be False (reseted the event flag)
```

Comment:
Event flag will be reset to `False` independently of the button state (pressed/released)
The flag is set to `True` by an event (interruption), that occurs when respective button's IO change from logic 1 to logic 0, and is se to `False` (cleared) whenever the `event()` function is called

Reply:
Test is good, just comments can be improved for better understanding of interrupt based buttons